dbt-snowflake/src/dbt/adapters/snowflake/parse_model.py

```diff
@@ -0,0 +1,98 @@
+from typing import Iterable, Optional
+
+from dbt_common.exceptions import DbtConfigError
+from dbt.adapters.contracts.relation import RelationConfig
+
+from dbt.adapters.snowflake import constants
+
+
+def automatic_clustering(model: RelationConfig) -> Optional[bool]:
+    if not model.config:
+        return None
+
+    return model.config.get("automatic_clustering", False)
+
+
+def base_location(model: RelationConfig) -> Optional[str]:
```

Comment:
this function is probably worth unit testing on it's own as it's now a lot easier to test